src/handler/mod.rs

```diff
@@ -891,41 +874,35 @@ impl Handler {
         }
     }
 
-    async fn send_next_request<P: ProtocolIdentity>(&mut self, node_address: NodeAddress) {
-        // ensure we are not over writing any existing requests
-        if self.active_requests.get(&node_address).is_none() {
-            if let std::collections::hash_map::Entry::Occupied(mut entry) =
-                self.pending_requests.entry(node_address)
+    async fn replay_active_requests<P: ProtocolIdentity>(&mut self, node_address: &NodeAddress) {
+        let active_requests = self
+            .active_requests
+            .remove_requests(node_address)
+            .unwrap_or_default();
+        for req in active_requests {
+            let request_id = req.id().clone();
+            if let Err(request_error) = self
+                .send_request::<P>(
+                    req.contact().clone(),
+                    request_id.clone(),
+                    req.body().clone(),
```

Comment:
there should be no need to clone the body and the contact, since the request call as a whole is dropped. It would be better to add a `into_request_parts(self)` (or similar) that gives the three parts that are necessary